Make auto_run's root registry key configurable

RoboticCheese opened the pull request:
I recently found myself working on a cookbook for an app that only installs on
a per-user basis, putting all its registry entries under the
`HKEY_CURRENT_USER` root key instead of `HKEY_LOCAL_MACHINE`.

This change adds a `root` property to the `windows_auto_run` resource,
maintaining the current behavior as a default and allowing support for
user-level applications as well.

Signed-off-by: Jonathan Hartman <j@hartman.io>

tas50 commented:
@RoboticCheese If you want to rebase this against master it would be nice to get merged in

tas50 commented:
@RoboticCheese Ping on a rebase here. I'd love to pull this in before we move this into chef-client so we can pull in this new functionality as well.

RoboticCheese commented:
Thanks for the reminder about this one. Rebased and updated. Seems to work again, though my Windows is pretty atrophied RN 🤞 

```
       Describing test::autorun
          Context windows_auto_run
           [+] does not auto-run Notepad for the machine 916ms
           [+] auto-runs Wordpad for the current user 95ms
       Tests completed in 1.01s
       Passed: 2 Failed: 0 Skipped: 0 Pending: 0
       Finished verifying <autorun-windows-10> (0m20.17s).
```

tas50 commented:
Thanks. I'll get this addition in when we move this into chef with Chef 14
